cvmfs/cvmfs.cc

```diff
@@ -311,7 +311,15 @@ static bool GetDirentForInode(const fuse_ino_t ino,
   if (ino == catalog_mgr->GetRootInode()) {
     bool retval =
       catalog_mgr->LookupPath(PathString(), catalog::kLookupDefault, dirent);
-    assert(retval);
+
+    std::string errormsg = "GetDirentForInode: Race condition? "
+                           "Not found dirent ";
+    errormsg.append(dirent->name().c_str());
```

Comment:
This triggers additional memory allocation(s) for the string concatenation. I'd prefer to do this only if `!retval`. Or, we make AssertOrLog a C variadic function like LogCvmfs, so that we can pass static C strings (mostly).

Actually, I think I prefer the second option because the code using it becomes nicer. See here: https://c-faq.com/varargs/handoff.html
We will need to split `LogCvmfs` into two functions, which is hopefully trivial.